### ISSUE TITLE
Can't run mriqc group analysis on a subset of the sample (separate echos)

### ISSUE THREAD

**Issue by caaam3** [nipreps/mriqc#1353]
### What happened?

I have multiecho data and I have ran mriqc on the participant level for all participants for each echo separately using bids-filter-file. Now I want to run a group analysis for each echo separately as I don't want all echos in the same group report. I have specified that in the bids-filter-file and when I run the code it identifies it as a filter and later on in the config file in logs it also only selects the first echo as intended. However, the generated reports are of all echos together.

### What command did you use?

```shell
docker run -it --rm -v <bids_root_dir>:/data:ro -v <output_dir>:/out \
-v <bids-filter-file_dir>:<bids-filter-file_dir> \
nipreps/mriqc:24.0.2 /data /out group \
--task-id nf \
--bids-filter-file <dir> \
--verbose-reports \
--no-sub \
-w <output_dir>/group/echo1
```

### What version of the software are you running?

24.1.0 and in the latest try 24.0.2 (see above) although it still shows 24.1.0. in logs

### How are you running this software?

Docker

### Is your data BIDS valid?

Yes

### Are you reusing any previously computed results?

No

**Comment by Remi-Gau** [comment]
I may be wrong but I think bids-filter are not yet supported at the group level, no?

Or more specifically you can pass a bids filter file but it won't affect the output.

**Comment by caaam3** [comment]
Yeah, that seems to be the case since it recognizes the filter file but it doesn't get used. I ended up having to separate the different echos into different folders in order to get separate group reports.

**Comment by Shotgunosine** [comment]
If I'm reading the code correctly, the bids filter would need to be applied in this step, where the json's are collected to construct a TSV:
https://github.com/nipreps/mriqc/blob/d7f9c89a227f8a4c2e7f721f405efa4f47cb6fde/mriqc/utils/misc.py#L199

Does that seem correct to folks more familiar with MRIQC?
